Backend/app/rider-platform/rider-app/Main/src/Domain/Action/Dashboard/CallbackRequest.hs

```diff
@@ -0,0 +1,78 @@
+{-
+ Copyright 2022-23, Juspay India Pvt Ltd
+
+ This program is free software: you can redistribute it and/or modify it under the terms of the GNU Affero General Public License
+
+ as published by the Free Software Foundation, either version 3 of the License, or (at your option) any later version. This program
+
+ is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+
+ or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero General Public License for more details. You should have received a copy of
+
+ the GNU Affero General Public License along with this program. If not, see <https://www.gnu.org/licenses/>.
+-}
+{-# OPTIONS_GHC -Wno-deprecations #-}
+{-# OPTIONS_GHC -Wno-incomplete-patterns #-}
+
+module Domain.Action.Dashboard.CallbackRequest where
+
+import Data.Time hiding (getCurrentTime)
+import qualified Domain.Types.CallbackRequest as DC
+import qualified Domain.Types.Merchant as DM
+import Environment
+import Kernel.External.Encryption
+import Kernel.Prelude
+import Kernel.Storage.Esqueleto hiding (count, isNothing)
+import Kernel.Types.APISuccess
+import Kernel.Types.Error
+import Kernel.Types.Id
+import Kernel.Utils.Common
+import qualified Storage.Queries.CallbackRequest as QCallbackRequest
+import qualified Storage.Queries.Merchant as QMerchant
+
+data CallbackRequestRes = CallbackRequestRes
+  { list :: [DC.CallbackRequestAPIEntity],
+    summary :: Summary
+  }
+  deriving (Show, Generic, ToJSON, FromJSON, ToSchema)
+
+data Summary = Summary
+  { totalCount :: Int,
+    count :: Int
+  }
+  deriving (Show, Generic, ToJSON, FromJSON, ToSchema)
+
+mobileIndianCode :: Text
+mobileIndianCode = "+91"
+
+getCallbackRequest :: ShortId DM.Merchant -> Maybe Int -> Maybe Int -> Maybe Text -> Maybe Text -> Maybe UTCTime -> Maybe DC.CallbackRequestStatus -> Flow CallbackRequestRes
+getCallbackRequest merchantShortId mbLimit mbOffset mbMobileCountryCode mbMobileNumber mbCreatedAt mbStatus = do
+  when (isJust mbMobileCountryCode && isNothing mbMobileNumber) $
+    throwError $ InvalidRequest "\"mobileCountryCode\" can be used only with \"mobileNumber\""
+  merchant <- runInReplica $ QMerchant.findByShortId merchantShortId >>= fromMaybeM (MerchantDoesNotExist merchantShortId.getShortId)
+  callbackRequest <- case (mbMobileNumber, mbStatus, mbCreatedAt) of
+    (Just mobileNumber, Nothing, Nothing) -> do
```

Comment:
Why at all we can't use more than one filter at once? Is it task requirements? Suppose, dashboard user wants to fetch all NEW callback requests from today, api will throw an error then

Reply:
If you copied this logic from `driverInfo` api, then in that api using more than one filter make no sense. Because one filter from list `(mbMobileNumber, mbVehicleNumber, mbDlNumber, mbRcNumber)` is enough to find specific driver

Reply:
resolved @roman-bodavskiy